src/Compilers/Test/Core/TargetFrameworkUtil.cs

```diff
@@ -261,5 +268,37 @@ string getName(MetadataReference m)
         public static IEnumerable<MetadataReference> GetReferencesWithout(TargetFramework targetFramework, params string[] excludeReferenceNames) =>
             GetReferences(targetFramework)
             .Where(x => !(x is PortableExecutableReference pe && excludeReferenceNames.Contains(pe.FilePath)));
+
+        /// <summary>
+        /// Many of our reference assemblies are only used by a subset of compiler unit tests. Having a PackageReference
+        /// to the assemblies here would cause them to be deployed to every unit test we write though. These are non-trivial 
+        /// in size (4+ MB) and we have ~50 test projects so this adds up fast. To keep size down we just add 
+        /// PackageReference on the few projects that and dynamically load here.
+        /// </summary>
+        private static ImmutableArray<PortableExecutableReference> LoadDynamicReferences(string assemblyName, string targetFrameworkName)
```

Comment:
>assemblyName [](http://example.com/codeflow?start=96&length=12)

Given that this method hardcodes the type name (minus the TFM name), consider doing the same for the assembly name. We can remove this parameter and use `var assemblyName = $"Basic.Reference.Assemblies.{targetFrameworkName}";`